packages/fluentui/react-northstar/src/components/Datepicker/Datepicker.tsx

```diff
@@ -225,6 +193,44 @@ export const Datepicker: ComponentWithAs<'div', DatepickerProps> &
     overrideProps: overrideDatepickerCalendarProps,
   });
 
+  const openStateToBooleanKnob = (openState: OpenState): boolean => {
+    return openState === OpenState.Open;
+  };
+
+  const onInputClick = (): void => {
+    if (openState === OpenState.Closed) {
+      setOpenState(OpenState.Open);
+    } else if (openState === OpenState.Open || openState === OpenState.Closing) {
+      // Keep popup open in case we can only enter the date through calendar.
+      if (props.allowTextInput) {
+        setOpenState(OpenState.Closed);
+      } else {
+        setOpenState(OpenState.Open);
+      }
+    }
+  };
+
+  const onInputChange = (e, target: { value: string }) => {
+    const parsedDate = props.parseDate(target.value);
+
+    setFormattedDate(target.value);
```

Comment:
Validation is pure and should be done outside of a component

Reply:
@layershifter just outside of the component's body or completely outside of this file?

Reply:
Probably even in a separate file. I expect something like this:

```
const { error } = validateDate()

setError(error)
```

Reply:
Updated in [the following commit](https://github.com/microsoft/fluentui/pull/14563/commits/f1dce86c84c5559c3e397d00f4b20bd814c3fb7f).